services/galley/test/integration/API/MLS.hs

```diff
@@ -1027,6 +1028,61 @@ testExternalCommitNewClient = do
 
 -- the list of members should be [alice1, bob1]
 
+-- | Check that external backend proposals are replayed after external commits
+-- AND that (external) client proposals are NOT replayed by the backend in the
+-- same case (since this is up to the clients).
+testExternalCommitNewClientResendBackendProposal :: TestM ()
+testExternalCommitNewClientResendBackendProposal = do
+  [alice, bob] <- createAndConnectUsers (replicate 2 Nothing)
+
+  runMLSTest $ do
+    [alice1, bob1, bob2] <- traverse createMLSClient [alice, bob, bob]
+    forM_ [bob1, bob2] uploadNewKeyPackage
+    (_, qcnv) <- setupMLSGroup alice1
+    void $ createAddCommit alice1 [bob] >>= sendAndConsumeCommitBundle
+    Just (_, kpBob2) <- find (\(ci, _) -> ci == bob2) <$> getClientsFromGroupState alice1 bob
+
+    mlsBracket [alice1, bob1] $ \[wsA, wsB] -> do
+      liftTest $
+        deleteClient (qUnqualified bob) (ciClient bob2) (Just defPassword)
+          !!! statusCode === const 200
+      WS.assertMatchN_ (5 # WS.Second) [wsB] $
+        wsAssertClientRemoved (ciClient bob2)
+
+      State.modify $ \mls ->
+        mls
+          { mlsMembers = Set.difference (mlsMembers mls) (Set.fromList [bob2])
+          }
```

Comment:
To me this deserves to be a test utility called e.g. `deleteMLSClient`.

Reply:
Would do that if we have more than the currently 2 occurrences of `deleteClient` plus state update.

Reply:
The main reason I proposed to put it in a function is to keep them together. Otherwise someone might forget to update the state after a call to `deleteClient`, and then hair-pulling debugging commences. Anyhow, this is just a suggestion.